### ISSUE TITLE
No animation when element hitting top edge of viewport

### ISSUE THREAD

**Issue by su-ex** [michalsnik/aos#389]
## This is:
Bug/Feature request/Question depending on how you look at it

## Specifications

  - AOS version: demo page
  - OS: Android 8.1 / 7.1.2 / probably all
  - Browser: Chrome / Firefox latest

## Expected Behavior
Animate element when scrolling down and element leaves viewport on top edge and other way around also of course when scrolling up and element comes into viewport on top edge.
Or maybe that's better to understand (read from beginning to end and end to beginning too): element invisible above viewport, animation on top edge, element visible inside viewport, animation on bottom edge (that currently solely is), element invisible below viewport

## Actual Behavior
Animation only on bottom edge

## Steps to Reproduce the Problem
Scrolling inside demo

## Possible Solution
If this is by intention add option to animate also on top edge.

**Comment by MarioPerini** [comment]
Did you try the option mirror:true in init?

**Comment by michalsnik** [comment]
Hi @su-ex! I believe what you're looking for is the `mirror` option, as @MarioPerini pointed. Check out documentation for more informations. If it's not - feel free to reopen this issue and tell us more about what are your expectations in regards to the mirror setting.

**Comment by su-ex** [comment]
Sorry, I've run out of time and forgot about this until I saw the notification bell on GitHub.
Yes, mirror option sounds good. I don't know how I could ooversee this although I actually read through all options.
But another question arrives out of that: With mirror top and bottom animation would be the same if I see it right, but if I want some sort of continuous scrolling page where elements fly in from top on the top side and from the bottom on the bottom side how could it be possible to achieve that (in the essence: how to set different animations per edge)?

**Comment by MarioPerini** [comment]
You can change the data-aos parameter with JS if you watch for the class aos-init and aos-animate.
EDIT:
or better change the data-aos parameter with the build in JS Event instead of watching classes
https://github.com/michalsnik/aos#js-events

```
document.addEventListener('aos:in', ({ detail }) => {
  console.log('animated in', detail);
});

document.addEventListener('aos:out', ({ detail }) => {
  console.log('animated out', detail);
});
```

**Comment by su-ex** [comment]
@MarioPerini
You mean detecting scrolling direction and then setting sort of animation dependant on that? At least that's the idea I got about that. But how would those events help me to achieve that? Also if I listen to those events, won't it already be too late to change the type of animation? Or when are those events triggered (before starting animation (what I would need) it after or while) or could the animation be deferred somehow till some stuff is done?

It's just that I'm unsure about all the corner cases meaning what is if it's detecting the scrolling direction exactly in the moment the AOS animation gets triggered, will it be too late to change it and the old one gets played? Hmm, I'll just experiment a bit with this at the end of the week (when I hopefully get some time for experiments 😋 ) and see what I can achieve. I'll try it with mirror: true, detect scrolling direction and set animation dependant on that (either by awaiting this in event or through changing the data-aos). Maybe it works well enough with ignoring some corner cases or this actually just works. Thanks for all your tips!
